api/provider.go

```diff
@@ -59,13 +61,14 @@ type PatchProviderRequest struct {
 }
 
 type UpdateProviderRequest struct {
-	ID           uid.ID                  `uri:"id" json:"-"`
-	Name         string                  `json:"name" example:"okta"`
-	URL          string                  `json:"url" example:"infrahq.okta.com"`
-	ClientID     string                  `json:"clientID" example:"0oapn0qwiQPiMIyR35d6"`
-	ClientSecret string                  `json:"clientSecret" example:"jmda5eG93ax3jMDxTGrbHd_TBGT6kgNZtrCugLbU"`
-	Kind         string                  `json:"kind" example:"oidc"`
-	API          *ProviderAPICredentials `json:"api"`
+	ID             uid.ID                  `uri:"id" json:"-"`
+	Name           string                  `json:"name" example:"okta"`
+	URL            string                  `json:"url" example:"infrahq.okta.com"`
+	ClientID       string                  `json:"clientID" example:"0oapn0qwiQPiMIyR35d6"`
+	ClientSecret   string                  `json:"clientSecret" example:"jmda5eG93ax3jMDxTGrbHd_TBGT6kgNZtrCugLbU"`
+	AllowedDomains []string                `json:"allowedDomains" example:"['example.com', 'infrahq.com']"`
```

Comment:
I think this needs another validation rule for `CreateProviderRequest` and `UpdateProviderRequest` to not allow commas (since we're using CommaSeparatedString, and maybe also set a maximum allowed length ? 

We could also require an `@` sign in every value, since that's needed later.

Reply:
Created StringSliceValidation